### DISCUSSION TITLE
Specify Gpu device

### DISCUSSION THREAD

**Discussion by JCNNBY** [AMReX-Codes/pyamrex#260] [Q&A] [answered]
Dear all,

I am using pyamrex with cuda backend. However, I cannot find where I can specify the gpu device_id.

Many thanks for your feedback.

Best

**Comment by WeiqunZhang** [comment, marked as answer]
For single process runs, we always use the first visible device. You can use CUDA_VISIBLE_DEVICES to control which devices are visible.

https://developer.nvidia.com/blog/cuda-pro-tip-control-gpu-visibility-cuda_visible_devices/

**Reply by ax3l** [reply]
Agreed. Alternatively, you can [switch the device context](https://docs.cupy.dev/en/stable/reference/generated/cupy.cuda.Device.html) with, e.g., cupy, and we will use it.

This this would be a bit advanced, because you want to do this before we do any device memory allocations in AMReX.
